apps/src/blockly/addons/functionEditor.js

```diff
@@ -153,15 +161,26 @@ export default class FunctionEditor {
    * procedure does not already exist.
    */
   showForFunction(procedure, procedureType) {
-    this.clearEditorWorkspace();
-
-    this.dom.style.display = 'block';
-    Blockly.common.svgResize(this.editorWorkspace);
-
     const existingProcedureBlock = Blockly.Procedures.getDefinition(
       procedure.getName(),
       Blockly.getHiddenDefinitionWorkspace()
     );
+    this.showForFunctionHelper(
+      existingProcedureBlock,
+      procedure,
+      procedureType
+    );
+  }
+
+  showForFunctionHelper(existingProcedureBlock, newProcedure, procedureType) {
```

Comment:
At this point, it feels like we way we handle the mutually exclusive existing/new procedures could be refactored for clarity (maybe something like make `showForFunction` determine whether to show an existing or a new procedure and two other functions `showForExistingFunction` and `showForNewFunction` to handle the specifics for each case, with commonalities extracted?). Should I file a ticket to propose this as a follow-up?

Reply:
I think that would be a good refactor, I don't feel super strongly either way.